autogoal/kb/_algorithm.py

```diff
@@ -339,8 +358,9 @@ def body(ns):
         ns["input_types"] = input_types_method
         ns["input_args"] = input_args_method
         ns["output_type"] = output_types_method
+        ns["get_inner_signature"] = get_inner_signature_method
 
-    return types.new_class(name=name, bases=(Algorithm,), exec_body=body)
+    return types.new_class(name=name, bases=(Algorithm, SeqWrapper), exec_body=body)
```

Comment:
Why do we need explicit inheritance here, if `get_inner_signature` is solved via introspection? Do you see any advantage in explicitly introducing a new class to the API (given that the introduced method is abstract, so we're not inheriting any functionality) vs just keeping it as an implicit convention?

Reply:
No advantage indeed. I probably tried solving it in a different way with this class but then just left it there afterwards.